### PR TITLE
Navigation: Fix `<Portal>` when `bodyScrolling` is enabled

### LINKED ISSUE #90927: RolePicker positioned incorrectly with `bodyScrolling`
> Regarding the bodyScrolling change, we are observing that the role picker behavior is broken, and it seems related to document layout. If you open the role picker (e.g. on the users list page), the frame for selecting roles is positioned after the main content. I'm doing what I can to triage the issue but am not sure whether a change to the role picker or to the other layout components is appropriate

Enterprise only

image

image

Looking at the RolePicker, it looks like it does a bunch of custom positioning, which probably is not longer correct when the bodyScrolling toggle is enabled.

We should refactor this to use Floating UI for this popover.

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- changes `<PortalContainer>` to have a `fixed` position so it positions correctly
- this change is behind the `bodyScrolling` feature toggle
- what happens when `bodyScrolling` is disabled:
  - the `PortalContainer` sits at the top of the document since `grafana-app` is positioned absolutely and therefore takes no space
  - if the page is scrolled whilst the portalled element is visible, the portalled element becomes detached from its anchor
- what happens now when `bodyScrolling` is enabled:
  - the `PortalContainer` sits at the top of the document since it has a fixed position
  - if the page is scrolled whilst the portalled element is visible, the portalled element becomes detached from it's anchor
  
tl;dr: same behaviour as before. @joshhunt is right in the issue description, we don't even really need `<Portal>` anymore since we can utilise floating-ui. but i'm aiming for parity first here before refactoring. 

**Why do we need this feature?**

- so `<Portal>` behaves correctly when using `bodyScrolling`

**Who is this feature for?**

- everyone

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/90927

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by joshhunt — APPROVED** [review]
good digging 👍
